packages/react/src/components/stepper/Step.tsx

```diff
@@ -0,0 +1,170 @@
+   */
+  label: string;
+  /**
+   * The language of the step
+   */
+  language?: Language;
+  /**
+   * The number of the step
+   */
+  number: number;
+  /**
+   * A callback function for custom action on step click
+   */
+  onStepClick?: (event, number: number) => void;
+  /**
+   * A function for rendering a custom aria label for step's state
+   */
+  renderCustomStateAriaLabel?: (step, state) => string;
+  /**
+   * A function for rendering a custom step count label
+   */
+  renderCustomStepCountLabel?: (step, totalNumberOfSteps) => string;
+  /**
+   * A boolean indicating whether step is in selected state
+   */
+  selected?: boolean;
+  /**
+   * A boolean indicating whether small variant is used
+   */
+  small?: boolean;
+  /**
+   * The state of the step
+   */
+  state: 'available' | 'completed' | 'disabled' | 'attention' | 'paused';
+  /**
+   * The total number of steps
+   */
+  stepsTotal: number;
+};
+
+type Language = 'en' | 'fi' | 'sv' | string;
+
+const getStepCountLabel = (language: Language, step: number, totalNumberOfSteps: number) => {
+  return {
+    en: `Step ${step}/${totalNumberOfSteps}.`,
+    fi: `Vaihe ${step}/${totalNumberOfSteps}.`,
+    sv: `Steg ${step}/${totalNumberOfSteps}.`,
+  }[language];
+};
+
+const states = {
+  available: {
+    fi: 'Valittavissa.',
+    en: 'Available.',
+    sv: 'Valbar.',
+  },
+  completed: {
+    fi: 'Valmis.',
+    en: 'Completed.',
+    sv: 'Komplett.',
+  },
+  attention: {
+    fi: 'Vaatii huomiota.',
+    en: 'Needs attention.',
+    sv: 'Behöver uppmärksamhet.',
+  },
+  paused: {
+    fi: 'Keskeytetty.',
+    en: 'Paused.',
+    sv: 'Pausad.',
+  },
+};
+
+const getStepState = (language: Language, state: StepProps['state']) => {
+  if (state === 'disabled') {
+    // Button has disabled attribute, which is enough for a11y
+    return '';
+  }
+  return states[state][language];
+};
+
+export const Step = React.forwardRef<HTMLButtonElement, StepProps>(
+  (
+    {
+      label,
+      language = 'fi',
+      number,
+      renderCustomStepCountLabel,
+      small = false,
+      state,
+      selected,
+      stepsTotal,
+      renderCustomStateAriaLabel,
+      onStepClick,
+      dataTestId,
+      ...rest
+    }: StepProps,
+    ref?: React.RefObject<HTMLButtonElement>,
+  ) => {
+    const composeAriaLabel = () => {
+      const stepCountLabel = renderCustomStepCountLabel
+        ? renderCustomStepCountLabel(number, stepsTotal)
+        : getStepCountLabel(language, number, stepsTotal);
+
+      let stateAriaLabel = renderCustomStateAriaLabel
+        ? renderCustomStateAriaLabel(number, state)
+        : getStepState(language, state);
+
+      if (selected && state === 'available') {
+        stateAriaLabel = '';
+      }
+
+      let labelWithPeriod = label;
+
+      if (labelWithPeriod.slice(-1) !== '.') {
+        labelWithPeriod += '.';
+      }
+
+      return [labelWithPeriod, stepCountLabel, stateAriaLabel].filter((lbl) => lbl).join(' ');
```

Comment:
I do not mind using label instead lbl 😁

Reply:
Unfortunately label is already defined in the upper scope